<img src="https://user-images.strikinglycdn.com/res/hrscywv4p/image/upload/c_limit,fl_lossy,h_300,w_300,f_auto,q_auto/1266110/Logo_wzxi0f.png" style="float: left; margin: 20px; height: 55px">

## Useful Operators

#### 1. Print first 5 numbers using range function

In [42]:
for i in range(5):
    print (i)

0
1
2
3
4


#### 2. Print integers within given start and stop number using range() function

In [41]:
for i in range(5, 10):
    print(i)

5
6
7
8
9


#### 3. Create a list of integers between 100 and 126 included

In [40]:
list(range(100,127))

[100,
 101,
 102,
 103,
 104,
 105,
 106,
 107,
 108,
 109,
 110,
 111,
 112,
 113,
 114,
 115,
 116,
 117,
 118,
 119,
 120,
 121,
 122,
 123,
 124,
 125,
 126]

#### 4.Print all odd numbers between 23 and 64 using range()

In [39]:
for i in range(23,64,2):
    print (i)

23
25
27
29
31
33
35
37
39
41
43
45
47
49
51
53
55
57
59
61
63


#### 5. Create a list with all the even numbers between 80 and 124 included.

In [38]:
list(range(80,126,2))

[80,
 82,
 84,
 86,
 88,
 90,
 92,
 94,
 96,
 98,
 100,
 102,
 104,
 106,
 108,
 110,
 112,
 114,
 116,
 118,
 120,
 122,
 124]

#### 6. Use enumerate to return the elemnts from the list where their index is greater than 3. Add them to a new list.

In [36]:
names = ['Pol','Javi','Pablo','Paco','Adri','Hector','Manu','Carlos','Ibrahim']

In [37]:
names_new = []
for i, k in enumerate(names):
    if i >= 4:
        names_new.append(k)
names_new

['Adri', 'Hector', 'Manu', 'Carlos', 'Ibrahim']

#### 7. Use enumerate to substract 2 to the numbers in the list that have and even index. Return the same list updated.

In [34]:
numbers = [2,44,564,0.3,45,6/7,9,15]

In [35]:
numbers_new = []
for i, k in enumerate(numbers):
    if i%2 == 0:
        numbers_new.append(k-2)
    else:
        numbers_new.append(k)

numbers_new

[0, 44, 562, 0.3, 43, 0.8571428571428571, 7, 15]

#### 8. Use zip to group each variable on the following list based on their index

In [32]:
list1 = [15,2.5,3/6,334,5]
list2 = ['t','n','o','a','e']

In [33]:
list(zip(list1,list2))

[(15, 't'), (2.5, 'n'), (0.5, 'o'), (334, 'a'), (5, 'e')]

#### 9. Write a program which returns does elements that appear in both lists. Add them to a new list and sort it descending.

In [ ]:
alpha = [6,7,33,6.5,908.9,34,2*2]
beta = [4,32,66,7,12+22]

In [13]:
alphabeta = []
for i in alpha:
    if i in beta:
        alphabeta.append(i)

sorted(alphabeta,reverse=True)

[34, 7, 4]

#### 10. Write a program which returns a list with does elements in the first list that match the length of the elements in the second list.

In [14]:
bv = ['Pol','Javi','Pablo','Paco','Adri','Hector','Manu','Carlos','Ibrahim','']
bv_num = [2,3,67,5,1,0]

In [15]:
newll = []
for i in bv_num:
    for k in bv:
        if i == len(k):
            newll.append(i)
            
newll

[3, 5, 0]

#### 11. Print a random number from a range between 100 and 4000.

In [16]:
import random
from random import randint
from random import shuffle
from random import randrange
from random import choice
from random import sample

In [17]:
randint(100,4000)

3286

#### 12. Print a shuffled version of the list provided.

In [18]:
numbers2 = [2,44,564,0.3,45,6/7,9,15]

In [19]:
shuffle(numbers2)
numbers2

[9, 564, 15, 0.8571428571428571, 2, 0.3, 44, 45]

#### 13. Print an even random number between 0 and 10.

In [20]:
randrange(0,10,2)

8

#### 14. Print a random element from a list

In [21]:
choice(['t','n','o','a','e','u','ff','adf'])

'ff'

#### 15. Print a sample of 4 numbers from the list provided

In [22]:
sample = [2,5,8,9,12,66,32,56,554]

In [23]:
random.sample(sample,4)

[12, 32, 9, 66]

#### 16. Create a program that asks for you name

In [24]:
input('Hi there, what is your name: ')

Hi there, what is your name:  Javier AKA Tabixe


'Javier AKA Tabixe'

#### 17. Create a program that asks for you name, age, location and returns the following output: Hi my name is _name_, I am _age_ years old and I live in _location_. Use any of the formatting you saw on day 1.

In [25]:
name = input("What is your name: ")
age = int(input("How old are you: "))
location = input("Where are you from: ")

print('Hi my name is {a}, I am {b} years old and I live in {c}'.format(a=name,b=age,c=location))

What is your name:  Tito Javi
How old are you:  23
Where are you from:  Santander


Hi my name is Tito Javi, I am 23 years old and I live in Santander


#### 18. Create a program that asks for you name, age and returns the year were you will be 100 years old. 

In [26]:
name = input("What is your name: ")
age = int(input("How old are you: "))
year = str((2019 - age)+100)
print(name + " will be 100 years old in the year " + year)

What is your name:  iVi
How old are you:  23


iVi will be 100 years old in the year 2096


#### 19. Write a single function called suma that returns the sum of the following examples withoug using more than one variable. Use *args

In [27]:
(3,5)
(4,5,6,7)
(1,2,3,5,6)

(1, 2, 3, 5, 6)

In [28]:
def suma(*num):
    sum = 0
    
    for n in num:
        sum = sum + n
    print("Sum:",sum)

In [29]:
suma(3,5)
suma(4,5,6,7)
suma(1,2,3,5,6)

Sum: 8
Sum: 22
Sum: 17


#### 20. Write a single function called multi_suma that returns the following. Use *args.

 - multi_suma(2,3,5) = ((2x3)+(2x5)) = 16
 - multi_suma(2,3,5,5) = ((2x3)+(2x5)+(2x5)) = 26
 - multi_suma(2,55,2,9) = ((2x55)+(2x2)+(2x9)) = 132

In [30]:
def multi_suma(a,*num):
    total = 0
    
    for n in num:
        total += n * a
    print("Sum:",total)

In [31]:
multi_suma(2,55,2,9)

Sum: 132
